include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,867 @@
+     * \param i The element to retrieve.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr reference operator[](size_type const i) noexcept
+    {
+        assert(i < size());
+        return data[i];
+    }
+
+    //!\copydoc operator[]()
+    constexpr const_reference operator[](size_type const i) const noexcept
+    {
+        assert(i < size());
+        return data[i];
+    }
+
+    /*!\brief Return the first element. Calling front on an empty container is undefined.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Calling front on an empty container is undefined. In debug mode an assertion checks the size of the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr reference front() noexcept
+    {
+        assert(size() > 0);
+        return (*this)[0];
+    }
+
+    //!\copydoc front()
+    constexpr const_reference front() const noexcept
+    {
+        assert(size() > 0);
+        return (*this)[0];
+    }
+
+    /*!\brief Return the last element.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Calling back on an empty container is undefined. In debug mode an assertion checks the size of the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr reference back() noexcept
+    {
+        assert(size() > 0);
+        return (*this)[size()-1];
+    }
+
+    //!\copydoc back()
+    constexpr const_reference back() const noexcept
+    {
+        assert(size() > 0);
+        return (*this)[size()-1];
+    }
+
+    /*!\name Capacity
+     * \{
+     */
+    /*!\brief Checks whether the container is empty.
+     * \returns `true` if the container is empty, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr bool empty() const noexcept
+    {
+        return size() == 0;
+    }
+
+    /*!\brief Returns the number of elements in the container, i.e. std::distance(begin(), end()).
+     * \returns The number of elements in the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr size_type size() const noexcept
+    {
+        return sz;
+    }
+
+    /*!\brief Returns the maximum number of elements the container is able to hold due to system or library
```

Comment:
Technically the docs are correct, but you have  a special case since you fix the max_size at compile time